### ISSUE TITLE
Installing with stack

### ISSUE THREAD

**Issue by jeffswitzer** [tidalcycles/tidal-midi#38]
I'm on mac os and have tidal installed and working via stack however I'm having trouble getting tidal-midi installed. 

Per the prereq instructions I did
`brew install portmidi`

Then when I ran `stack install tidal-midi` I got the following message:
```
In the dependencies for tidal-midi-0.9.4:
    PortMidi must match ==0.1.6.0, but the stack configuration has no specified version (latest applicable is 0.1.6.0)
    tidal-0.1.0.0 must match >=0.9.4 (latest applicable is 0.9.4)

Recommended action: try adding the following to your extra-deps in /Users/jeff/Development/tidal/stack.yaml:
- PortMidi-0.1.6.0
```

So I added - PortMidi-0.1.6.0 to my stack.yaml

Then when I run stack install tidal-midi I still get:
```

Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for tidal-midi-0.9.4:
    tidal-0.1.0.0 must match >=0.9.4 (latest applicable is 0.9.4)

Plan construction failed.

```

Any tips? I'm new to haskell, but heard using stack was the better route than cabal. I'm used to npm in the js world though and it seems similar.

I also tried to install 0.9.4 specifically but got the following:
```
kaladin:tidal jeff$ stack install tidal-0.9.4
Error parsing targets: tidal target has a specific version number, but it is a local package.
To avoid confusion, we will not install the specified version or build the local one.
To build the local package, specify the target without an explicit version.
```
Thanks for any help!

**Comment by kindohm** [comment]
I don't use MacOS, but from the tidal-midi prerequisites:
https://github.com/tidalcycles/tidal-midi#prerequisites

It says you need to run "brew install portmidi" first. Have you tried
that?

Aside: I just realized I forgot to mention this step for MacOS users in
my tidal-midi videos :(

-Mike

**Comment by jeffswitzer** [comment]
@kindohm yep I've ran `brew install portmidi`
This error is weird to me:
`tidal-0.1.0.0 must match >=0.9.4 (latest applicable is 0.9.4)`
Do you think there is a typo in the most recent tidal stack version, like it should be 1.0.0 instead of 0.1.0.0?

**Comment by kindohm** [comment]
Jeff,

I've reproduced the problem on Windows, and got up to the point where
you see the Tidal version error. However, I see a different version
number:

Error: While constructing the build plan, the following exceptions were
encountered:

In the dependencies for tidal-midi-0.9.4:
    tidal-0.8.2 must match >=0.9.4 (latest applicable is 0.9.4)

I see 0.8.2 instead of 0.1.0.0. From what I can tell, Tidal-0.8.2 is the
version that has been published to Stackage. I guess the next step I'd
like to take is to understand how to get Tidal-0.9.4 published to
Stackage, but I'm not sure if that will solve your problem since you're
seeing "0.1.0.0". Not sure what is going on there.


-- 
  Mike Hodnick
  mike@kindohm.com

**Comment by jeffswitzer** [comment]
Since you are seeing 0.8.2 which is the current tidal version in stackage LTS, you can switch to the stackage nightly build to get tidal-0.9.4. I was messing with this last night - here is a related [stackoverflow](http://stackoverflow.com/questions/44014536/ask-haskell-stack-to-list-available-versions-of-a-package) I opened. I didn't get it all figured out yet but will let you know as I progress. Do you know how to answer Alexis Kings question in the stackoverflow about whether I need to be doing `stack install` vs `stack build`? I'm trying to use `stack exec atom` at the end to open the atom editor which is supposed to allow the atom tidal plugin to use stacks ghci I think. At one point I had that part working just without the tidal-midi support.

**Comment by kindohm** [comment]
Yes, I got "stack exec atom" to work, and Tidal/atom will use stack's
ghci. I'll look more into the tidal-midi issue and the stackoverflow
issue later.

-Mike


-- 
  Mike Hodnick
  mike@kindohm.com
